Simplify bug report template (#17561)
mistercrunch/Airflow: .github/ISSUE_TEMPLATE/bug_report.md

---

<!--
Welcome to Apache Airflow!

Please complete the next sections or the issue will be closed.
-->

**Apache Airflow version**:

<!-- AIRFLOW VERSION IS MANDATORY -->

**OS**:

<!-- MANDATORY! You can get it via `cat /etc/oss-release` for example -->

**Apache Airflow Provider versions**:

<!-- You can use `pip freeze | grep apache-airflow-providers` (you can leave only relevant ones)-->

> **Deployment: Virtualenv/ VM / Docker-compose / K8S / Helm Chart / Managed Airflow Service**:

Reviewer: ```suggestion
**Deployment**:

<!-- e.g. Virtualenv / VM / Docker-compose / K8S / Helm Chart / Managed Airflow Service -->
```

> <!-- Please include version your deployment tools: docker-compose, k8s, helm, etc -->

Reviewer: ```suggestion
<!-- Please include your deployment tools and versions: docker-compose, k8s, helm, etc -->
```

---

<!--
Welcome to Apache Airflow!

Please complete the next sections or the issue will be closed.
-->

**Apache Airflow version**:

<!-- AIRFLOW VERSION IS MANDATORY -->

> **OS**:

Reviewer: I think we should flip the order of OS and provider versions.

Author: Why? I actually deliberately replaced them, thinking that more often that not the problems will be connected with the OS than provider. When I think about an issue raised it's 

`I am running Airflow on MacOS and I have those providers installed`.  The "MacOS" is often much more important which providers are installed. I think providers are important in very small percentage of issues :)

Reviewer: To me it makes sense to have airflow version info and os/deployment info grouped 🤷‍♂️. I'm not picky though.

**Apache Airflow Provider versions**:

<!-- You can use `pip freeze | grep apache-airflow-providers` (you can leave only relevant ones)-->

**Deployment**:

<!-- e.g. Virtualenv / VM / Docker-compose / K8S / Helm Chart / Managed Airflow Service -->

<!-- Please include your deployment tools and versions: docker-compose, k8s, helm, etc -->

**What happened**:

<!-- Please include exact error messages if you can -->

**What you expected to happen**:

<!-- What do you think went wrong? -->

**How to reproduce it**:

<!--
As minimally and precisely as possible. Keep in mind we do not have access to your cluster or dags.
If this is a UI bug, please provide a screenshot of the bug or a link to a youtube video of the bug in action
You can include images/screen-casts etc. by drag-dropping the image here.
-->

**Anything else we need to know**:

<!--
How often does this problem occur? Once? Every time etc?
Any relevant logs to include? Put them here inside fenced
``` ``` blocks or inside a foldable details tag if it's long:
<details><summary>x.log</summary> lots of stuff </details>
-->

**Are you willing to submit a PR?**

<!---

> This is not required, but we are happy to guide you in contribution process

Reviewer: ```suggestion
This is absolutely not required, but we are happy to guide you in contribution process
```

especially if you already have a good understanding of how to implement the fix.

> Airflow is a community-managed project and we love to bring new contributors in.

Reviewer: ```suggestion
Airflow is a community-managed project and we love to bring new contributors in.
Find us in #airflow-how-to-pr on Slack!
```
